src/routes/protected-routes.tsx

```diff
@@ -1,9 +1,20 @@
+import { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
 import { Navigate, Outlet } from 'react-router';
 
-import { useAuth } from '@/api/queries/useAuth';
+import { useAuth } from '@/api/queries/auth/useAuth';
+import { useGetProfile } from '@/api/queries/profile';
 
 export default function ProtectedRoute() {
+  const { i18n } = useTranslation();
   const { data: isAuthenticated, isError } = useAuth();
+  const { data: profile } = useGetProfile();
+
+  useEffect(() => {
+    if (profile?.locale) {
+      i18n.changeLanguage(profile.locale);
+    }
+  }, [profile?.locale, i18n]);
```

Comment:
profile의 locale에 따라서 변하는 값이니깐 useEffectEvent로 조건을 명확하게 해야해요. 이런식으로.
```suggestion
  const changeLocaleEffect = useEffectEvent((locale: string) => {
      i18n.changeLanguage(locale);
  })
  useEffect(() => {
    if (profile?.locale) {
      changeLocaleEffect(profile.locale);
    }
  }, [profile?.locale]);
```